Lib/html/parser.py

```diff
@@ -260,7 +260,7 @@ def parse_html_declaration(self, i):
         if rawdata[i:i+4] == '<!--':
             # this case is actually already handled in goahead()
             return self.parse_comment(i)
-        elif rawdata[i:i+3] == '<![':
+        elif rawdata[i:i+3] == '<![CDATA[':
```

Comment:
`rawdata[i:i+9]`

Reply:
I updated this and added a couple of tests for good measure.